src/driver/cockroachdb/CockroachDriver.ts

```diff
@@ -614,17 +621,43 @@ export class CockroachDriver implements Driver {
             // console.log("isGenerated:", tableColumn.isGenerated, columnMetadata.isGenerated);
             // console.log("==========================================");
 
-            return tableColumn.name !== columnMetadata.databaseName
-                || tableColumn.type !== this.normalizeType(columnMetadata)
-                || tableColumn.length !== columnMetadata.length
-                || tableColumn.precision !== columnMetadata.precision
-                || (columnMetadata.scale !== undefined && tableColumn.scale !== columnMetadata.scale)
-                || (tableColumn.comment || "") !== columnMetadata.comment
-                || (!tableColumn.isGenerated && this.lowerDefaultValueIfNecessary(this.normalizeDefault(columnMetadata)) !== tableColumn.default) // we included check for generated here, because generated columns already can have default values
-                || tableColumn.isPrimary !== columnMetadata.isPrimary
-                || tableColumn.isNullable !== columnMetadata.isNullable
-                || tableColumn.isUnique !== this.normalizeIsUnique(columnMetadata)
-                || tableColumn.isGenerated !== columnMetadata.isGenerated;
+            // Pulled out individual conditions for easier debugging.
+
+            const nameCond = tableColumn.name !== columnMetadata.databaseName;
+            const typeCond = tableColumn.type !== this.normalizeType(columnMetadata);
+            const lengthCond = tableColumn.length !== columnMetadata.length;
+
+            // Intentionally using '!=' instead of '!==' so that undefined and null compare as equal.
+            const precisionCond = tableColumn.precision != columnMetadata.precision;
```

Comment:
not sure I really like `!=` approach instead of explicit check for null or undefined, because `!=` forces me to ask questions:

* what if tableColumn.precision is `0` and columnMetadata.precision is `undefined` or `null`? What behaviour should I expect?
* why the hell precision can be both `null` or `undefined`? In what situations it can be `null` and in what situations it can be `undefined`? Feels like it should be only one of them when the value isn't defined? Maybe normalization can be applied somewhere else before this check?

Reply:
same questions can arise in different heads, and instead of writing the comments maybe it's better to replace this check with explicit one?

Reply:
If we could bump the TypeScript version a nice check here would be
```suggestion
            const precisionCond = (tableColumn.precision ?? null) !== (columnMetadata.precision ?? null);
```

Reply:
This is now possible on latest TypeScript 😀